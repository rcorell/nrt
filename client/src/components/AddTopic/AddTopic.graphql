mutation CreateTopic($groupId: ID!, $title: String!, $description: String) {
        createTopic(groupId: $groupId, title: $title, description: $description) {
            id
        }
    }