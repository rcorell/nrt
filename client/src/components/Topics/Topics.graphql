query FetchTopics {
        topics {
            id
            title
            description
            group {
                id
                name
            }
        }
    }