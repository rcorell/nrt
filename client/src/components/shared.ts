export const LOADING_TEXT = 'Loading...';
