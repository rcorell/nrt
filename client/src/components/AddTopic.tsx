import { useMutation, useQuery } from '@apollo/client';
import React, { useState } from 'react';
import { Button, Form, FormControlProps } from 'react-bootstrap';

import { CreateTopicMutation, CreateTopicMutationVariables } from 'src/api/__generated__/CreateTopicMutation';
import { FetchUserQuery } from 'src/api/__generated__/FetchUserQuery';
import { createTopicMutationString, fetchUserQueryString } from 'src/api/api';
import { LOADING_TEXT } from 'src/components/shared';
import { AppError, AppForm, FormContainer } from 'src/styles/form';
import { setBrowserTitle } from 'src/utils';

export const AddTopic: React.FC = () => {
    const [description, setDescription] = useState('');
    const [groupId, setGroupId] = useState('');
    const [title, setTitle] = useState('');

    const userQuery = useQuery<FetchUserQuery>(fetchUserQueryString);

    const [createTopic, { error }] = useMutation<CreateTopicMutation, CreateTopicMutationVariables>(
        createTopicMutationString,
        {
            variables: { groupId, title, description },
            onError: () => {
                // RTL bug
            }
        }
    );

    const isFormInvalid = () => {
        return title.length < 1;
    };

    const handleSubmit = (event: React.FormEvent<FormControlProps>) => {
        event.preventDefault();

        createTopic();
    };

    const errorStatus = () => {
        if (error) {
            return <div>Error: ${JSON.stringify(error)}</div>;
        }

        return null;
    };

    setBrowserTitle('Add Topic');

    if (userQuery.loading || !userQuery.data) {
        return <div>{LOADING_TEXT}</div>;
    }

    const groupOptions = userQuery.data.user.groups.map((group) => (
        <option key={group!.id} value={group!.id}>
            {group!.name}
        </option>
    ));

    return (
        <FormContainer>
            <h1>Add Topic</h1>
            {errorStatus()}
            <AppForm onSubmit={handleSubmit}>
                <Form.Group>
                    <Form.Label id="groupLabel">Group</Form.Label>
                    <Form.Control
                        as="select"
                        aria-labelledby="groupLabel"
                        id="group"
                        onChange={(e) => setGroupId(e.target.value)}
                        size="lg"
                    >
                        {groupOptions}
                    </Form.Control>
                </Form.Group>

                <Form.Group>
                    <Form.Label id="topicLabel">Topic</Form.Label>
                    <Form.Control
                        autoFocus
                        aria-labelledby="topicLabel"
                        id="topic"
                        onChange={(e) => setTitle(e.target.value)}
                        size="lg"
                        value={title}
                    />
                </Form.Group>

                <Form.Group>
                    <Form.Label id="descriptionLabel">Description</Form.Label>
                    <Form.Control
                        id="description"
                        aria-labelledby="descriptionLabel"
                        onChange={(e) => setDescription(e.target.value)}
                        size="lg"
                        value={description}
                    />
                </Form.Group>

                <Button type="submit" disabled={isFormInvalid()}>
                    Submit Topic
                </Button>
            </AppForm>
            <AppError></AppError>
        </FormContainer>
    );
};
