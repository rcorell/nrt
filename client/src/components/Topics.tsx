import { useQuery } from '@apollo/client';
import _ from 'lodash';
import MaterialTable from 'material-table';
import React from 'react';

import { FetchTopicsQuery } from 'src/api/__generated__/FetchTopicsQuery';
import { fetchTopicsQuery } from 'src/api/api';
import { LOADING_TEXT } from 'src/components/shared';
import { setBrowserTitle } from 'src/utils';

const COLUMN_DEFINITIONS = [
    { field: 'title', title: 'Title' },
    { field: 'description', title: 'Description' },
    { field: 'group', title: 'Group' }
];

export const Topics: React.FC = () => {
    setBrowserTitle('Topics');

    const { data, error, loading } = useQuery<FetchTopicsQuery>(fetchTopicsQuery, { fetchPolicy: 'network-only' });

    if (loading) {
        return <div>{LOADING_TEXT}</div>;
    }

    if (error) {
        return <div>Error: {JSON.stringify(error)}</div>;
    }

    const topics = _.cloneDeep(data!.topics).map((topic) => ({ ...topic, group: topic.group.name }));

    return (
        <MaterialTable
            title="Topics"
            columns={COLUMN_DEFINITIONS}
            data={topics}
            options={{
                pageSize: 20,
                pageSizeOptions: [20, 50, 100]
            }}
        />
    );
};
