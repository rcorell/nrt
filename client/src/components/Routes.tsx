import { createBrowserHistory } from 'history';
import React from 'react';
import { Route, Router, Switch } from 'react-router-dom';

import { AddGroup } from 'src/components/AddGroup';
import { AddTopic } from 'src/components/AddTopic';
import { Home } from 'src/components/Home';
import { Login } from 'src/components/Login';
import { NotFound } from 'src/components/NotFound';
import { Path } from 'src/redux/navigation/types';
import { Signup } from 'src/components/Signup';
import { Topics } from 'src/components/Topics';

import { NavigationBar } from './NavigationBar';
import { PrivateRoute } from './PrivateRoute';

export const history = createBrowserHistory();

export const Routes: React.FC = () => {
    return (
        <Router history={history}>
            <NavigationBar />

            <Switch>
                <Route exact path={Path.LOGIN} component={Login} />
                <Route exact path={Path.SIGNUP} component={Signup} />
                <PrivateRoute exact path={Path.ADD_TOPIC} component={AddTopic} />
                <PrivateRoute exact path={Path.TOPICS} component={Topics} />
                <PrivateRoute exact path={Path.ADD_GROUP} component={AddGroup} />
                <PrivateRoute exact path={Path.HOME} component={Home} />

                <Route component={NotFound} />
            </Switch>
        </Router>
    );
};
