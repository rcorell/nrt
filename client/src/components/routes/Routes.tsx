import { createBrowserHistory } from 'history';
import * as React from 'react';
import { connect } from 'react-redux';
import { Route, Router, Switch } from 'react-router-dom';

import { AddTopic } from 'src/components/addTopic/AddTopic';
import { Home } from 'src/components/home/Home';
import { Login } from 'src/components/login/Login';
import { Path } from 'src/redux/navigation/types';
import { Signup } from 'src/components/signup/Signup';
import { Topics } from 'src/components/topics/Topics';

import { NavigationBar } from './NavigationBar';
import { PrivateRoute } from './PrivateRoute';

const history = createBrowserHistory();
export { history };

export class RoutesComponent extends React.Component<{}> {
    public constructor(props: any) {
        super(props);
    }

    render() {
        return (
            <Router history={history}>
                <NavigationBar />

                <Switch>
                    <Route exact path={Path.LOGIN} component={Login} />
                    <Route exact path={Path.SIGNUP} component={Signup} />
                    <Route exact path={Path.ADD_TOPIC} component={AddTopic} />
                    <Route exact path={Path.TOPICS} component={Topics} />

                    {/* <Route exact path={Path.OUT_OF_SERVICE} component={OutOfService} />
                    <Route exact path={Path.WAITING} component={Waiting} /> */}

                    <PrivateRoute exact path={Path.HOME} component={Home} />
                </Switch>
            </Router>
        );
    }
}

export const Routes = connect()(RoutesComponent);
