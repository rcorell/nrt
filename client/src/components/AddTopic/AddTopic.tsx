import { useMutation, useQuery } from '@apollo/client';
import { navigate } from 'hookrouter';
import React, { useState } from 'react';
import { Button, Form, FormControlProps } from 'react-bootstrap';

import { CreateTopicMutation, CreateTopicMutationVariables } from 'src/api/__generated__/CreateTopicMutation';
import { FetchUserQuery } from 'src/api/__generated__/FetchUserQuery';
import { createTopicMutation, fetchUserQuery } from 'src/api/api';
import { Path } from 'src/components/Routes';
import { LOADING_TEXT } from 'src/components/shared';
import { AppError, AppForm, FormContainer } from 'src/styles/form';
import { setBrowserTitle } from 'src/utils';

export const AddTopic: React.FC = () => {
    const [description, setDescription] = useState('');
    const [groupId, setGroupId] = useState('');
    const [title, setTitle] = useState('');

    const userQuery = useQuery<FetchUserQuery>(fetchUserQuery, { fetchPolicy: 'network-only' });

    const [createTopic, { error }] = useMutation<CreateTopicMutation, CreateTopicMutationVariables>(
        createTopicMutation,
        {
            onCompleted: () => {
                navigate(Path.TOPICS);
            },
            onError: () => {
                // RTL bug
            },
            variables: { description, groupId, title }
        }
    );

    const isFormInvalid = () => {
        return title.length < 1;
    };

    const handleSubmit = (event: React.FormEvent<FormControlProps>) => {
        event.preventDefault();

        createTopic();
    };

    const errorStatus = () => {
        if (error) {
            return <div>Error: ${JSON.stringify(error)}</div>;
        }

        return null;
    };

    setBrowserTitle('Add Topic');

    if (userQuery.loading || !userQuery.data) {
        return <div>{LOADING_TEXT}</div>;
    }

    if (userQuery.data.user.groups.length === 0) {
        return <div>You must join a group to add topics.</div>;
    }

    const groupOptions = userQuery.data.user.groups.map((group) => (
        <option key={group!.id} value={group!.id}>
            {group!.name}
        </option>
    ));

    if (groupId === '') {
        setGroupId(userQuery.data.user.groups[0]!.id);
    }

    return (
        <FormContainer>
            <h1>Add Topic</h1>
            {errorStatus()}
            <AppForm onSubmit={handleSubmit}>
                <Form.Group>
                    <Form.Label id="groupLabel">Group</Form.Label>
                    <Form.Control
                        as="select"
                        aria-labelledby="groupLabel"
                        defaultValue={groupId}
                        id="group"
                        onChange={(e) => {
                            setGroupId(e.target.value);
                        }}
                        size="lg"
                    >
                        {groupOptions}
                    </Form.Control>
                </Form.Group>

                <Form.Group>
                    <Form.Label id="topicLabel">Topic</Form.Label>
                    <Form.Control
                        autoFocus
                        aria-labelledby="topicLabel"
                        id="topic"
                        onChange={(e) => setTitle(e.target.value)}
                        size="lg"
                        value={title}
                    />
                </Form.Group>

                <Form.Group>
                    <Form.Label id="descriptionLabel">Description</Form.Label>
                    <Form.Control
                        id="description"
                        aria-labelledby="descriptionLabel"
                        onChange={(e) => setDescription(e.target.value)}
                        size="lg"
                        value={description}
                    />
                </Form.Group>

                <Button type="submit" disabled={isFormInvalid()}>
                    Submit Topic
                </Button>
            </AppForm>
            <AppError></AppError>
        </FormContainer>
    );
};
