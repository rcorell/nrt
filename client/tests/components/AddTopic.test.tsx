import { MockedResponse } from '@apollo/client/testing';
import { fireEvent, screen, waitForElementToBeRemoved } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { AddTopic } from 'src/components/AddTopic/AddTopic';
import { Path } from 'src/components/Routes';
import { LOADING_TEXT } from 'src/components/shared';
import { ADD_TOPIC, INVALID, VALID } from 'tests/fixtures';
import { mockGroups } from 'tests/mocks/groupMocks';
import { fetchUserWithGroupsMocks, fetchUserWithoutGroupsMocks } from 'tests/mocks/userMocks';
import { lastNavigationPath, oneTick, renderComponent, setField, testFormSnapshots } from 'tests/testHelpers';

import { createTopicMocks } from './Topics/Topics.mocks';

describe('AddTopic', () => {
    testFormSnapshots(
        {
            component: AddTopic,
            fields: [
                {
                    invalidValue: INVALID.TOPIC.DESCRIPTION,
                    label: ADD_TOPIC.DESCRIPTION,
                    validValue: VALID.TOPIC.DESCRIPTION
                },
                { invalidValue: INVALID.TOPIC.TITLE, label: ADD_TOPIC.TITLE, validValue: VALID.TOPIC.TITLE }
            ],
            pageName: ADD_TOPIC.PAGE_NAME
        },
        [fetchUserWithGroupsMocks.success]
    );

    const submitAddTopic = async (mocks: MockedResponse[]) => {
        const { container } = renderComponent(AddTopic, mocks);

        await waitForElementToBeRemoved(() => screen.getByText(LOADING_TEXT));
        userEvent.selectOptions(screen.getByLabelText('Group'), mockGroups[0].id);
        setField(ADD_TOPIC.TITLE, VALID.TOPIC.TITLE);
        setField(ADD_TOPIC.DESCRIPTION, VALID.TOPIC.DESCRIPTION);
        const submitButton = screen.getByRole('button');
        fireEvent.click(submitButton);
        await oneTick();

        return container;
    };

    describe('success', () => {
        describe('with groups', () => {
            it('should create a topic and navigate to the topics page', async () => {
                const container = await submitAddTopic([
                    fetchUserWithGroupsMocks.success,
                    createTopicMocks.success
                ]);

                expect(container).toMatchSnapshot();
                expect(createTopicMocks.success.newData).toHaveBeenCalledTimes(1);
                expect(lastNavigationPath).toEqual(Path.TOPICS);
            });
        });

        describe('without groups', () => {
            it('should show a helpful message', async () => {
                const { container } = renderComponent(AddTopic, [fetchUserWithoutGroupsMocks.success]);

                await oneTick();

                expect(container).toMatchSnapshot();
            });
        });
    });

    describe('failure', () => {
        it('createTopic: network error', async () => {
            const container = await submitAddTopic([
                fetchUserWithGroupsMocks.success,
                createTopicMocks.networkError
            ]);

            expect(container).toMatchSnapshot();
            expect(screen.queryByText(/CreateTopic: network error/)).toBeInTheDocument();
        });

        it('createTopic: GraphQL error', async () => {
            const container = await submitAddTopic([
                fetchUserWithGroupsMocks.success,
                createTopicMocks.graphQLError
            ]);

            expect(container).toMatchSnapshot();
            expect(screen.queryByText(/CreateTopic: GraphQL error/)).toBeInTheDocument();
        });
    });
});
