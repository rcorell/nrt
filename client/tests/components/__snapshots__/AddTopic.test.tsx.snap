// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTopic failure createTopic: GraphQL error 1`] = `
<div>
  <div
    class="sc-bdvvaa hMSAon"
  >
    <h1>
      Add Topic
    </h1>
    <div>
      Error: $
      {"graphQLErrors":[{"message":"CreateTopicMutation: GraphQL error"}],"networkError":null,"message":"CreateTopicMutation: GraphQL error"}
    </div>
    <form
      class="sc-gsDJrp idIFaR"
    >
      <div
        class="form-group"
      >
        <label
          class="form-label"
          id="groupLabel"
        >
          Group
        </label>
        <select
          aria-labelledby="groupLabel"
          class="form-control form-control-lg"
          id="group"
        >
          <option
            selected=""
            value="40"
          >
            Blue Blaze Irregulars
          </option>
        </select>
      </div>
      <div
        class="form-group"
      >
        <label
          class="form-label"
          id="topicLabel"
        >
          Topic
        </label>
        <input
          aria-labelledby="topicLabel"
          class="form-control form-control-lg"
          id="topic"
          value="A Really Great Topic"
        />
      </div>
      <div
        class="form-group"
      >
        <label
          class="form-label"
          id="descriptionLabel"
        >
          Description
        </label>
        <input
          aria-labelledby="descriptionLabel"
          class="form-control form-control-lg"
          id="description"
          value="A really great topic description for a really great topic."
        />
      </div>
      <button
        class="btn btn-primary"
        type="submit"
      >
        Submit Topic
      </button>
    </form>
    <div
      class="sc-dkPtyc cWQCqc"
    />
  </div>
</div>
`;

exports[`AddTopic failure createTopic: network error 1`] = `
<div>
  <div
    class="sc-bdvvaa hMSAon"
  >
    <h1>
      Add Topic
    </h1>
    <div>
      Error: $
      {"graphQLErrors":[],"networkError":{},"message":"CreateTopicMutation: network error"}
    </div>
    <form
      class="sc-gsDJrp idIFaR"
    >
      <div
        class="form-group"
      >
        <label
          class="form-label"
          id="groupLabel"
        >
          Group
        </label>
        <select
          aria-labelledby="groupLabel"
          class="form-control form-control-lg"
          id="group"
        >
          <option
            selected=""
            value="40"
          >
            Blue Blaze Irregulars
          </option>
        </select>
      </div>
      <div
        class="form-group"
      >
        <label
          class="form-label"
          id="topicLabel"
        >
          Topic
        </label>
        <input
          aria-labelledby="topicLabel"
          class="form-control form-control-lg"
          id="topic"
          value="A Really Great Topic"
        />
      </div>
      <div
        class="form-group"
      >
        <label
          class="form-label"
          id="descriptionLabel"
        >
          Description
        </label>
        <input
          aria-labelledby="descriptionLabel"
          class="form-control form-control-lg"
          id="description"
          value="A really great topic description for a really great topic."
        />
      </div>
      <button
        class="btn btn-primary"
        type="submit"
      >
        Submit Topic
      </button>
    </form>
    <div
      class="sc-dkPtyc cWQCqc"
    />
  </div>
</div>
`;

exports[`AddTopic standard snapshots fresh load 1`] = `
<div>
  <div
    class="sc-bdvvaa hMSAon"
  >
    <h1>
      Add Topic
    </h1>
    <form
      class="sc-gsDJrp idIFaR"
    >
      <div
        class="form-group"
      >
        <label
          class="form-label"
          id="groupLabel"
        >
          Group
        </label>
        <select
          aria-labelledby="groupLabel"
          class="form-control form-control-lg"
          id="group"
        >
          <option
            selected=""
            value="40"
          >
            Blue Blaze Irregulars
          </option>
        </select>
      </div>
      <div
        class="form-group"
      >
        <label
          class="form-label"
          id="topicLabel"
        >
          Topic
        </label>
        <input
          aria-labelledby="topicLabel"
          class="form-control form-control-lg"
          id="topic"
          value=""
        />
      </div>
      <div
        class="form-group"
      >
        <label
          class="form-label"
          id="descriptionLabel"
        >
          Description
        </label>
        <input
          aria-labelledby="descriptionLabel"
          class="form-control form-control-lg"
          id="description"
          value=""
        />
      </div>
      <button
        class="btn btn-primary"
        disabled=""
        type="submit"
      >
        Submit Topic
      </button>
    </form>
    <div
      class="sc-dkPtyc cWQCqc"
    />
  </div>
</div>
`;

exports[`AddTopic standard snapshots invalid parameters 1`] = `
<div>
  <div
    class="sc-bdvvaa hMSAon"
  >
    <h1>
      Add Topic
    </h1>
    <form
      class="sc-gsDJrp idIFaR"
    >
      <div
        class="form-group"
      >
        <label
          class="form-label"
          id="groupLabel"
        >
          Group
        </label>
        <select
          aria-labelledby="groupLabel"
          class="form-control form-control-lg"
          id="group"
        >
          <option
            selected=""
            value="40"
          >
            Blue Blaze Irregulars
          </option>
        </select>
      </div>
      <div
        class="form-group"
      >
        <label
          class="form-label"
          id="topicLabel"
        >
          Topic
        </label>
        <input
          aria-labelledby="topicLabel"
          class="form-control form-control-lg"
          id="topic"
          value=""
        />
      </div>
      <div
        class="form-group"
      >
        <label
          class="form-label"
          id="descriptionLabel"
        >
          Description
        </label>
        <input
          aria-labelledby="descriptionLabel"
          class="form-control form-control-lg"
          id="description"
          value=""
        />
      </div>
      <button
        class="btn btn-primary"
        disabled=""
        type="submit"
      >
        Submit Topic
      </button>
    </form>
    <div
      class="sc-dkPtyc cWQCqc"
    />
  </div>
</div>
`;

exports[`AddTopic standard snapshots valid parameters 1`] = `
<div>
  <div
    class="sc-bdvvaa hMSAon"
  >
    <h1>
      Add Topic
    </h1>
    <form
      class="sc-gsDJrp idIFaR"
    >
      <div
        class="form-group"
      >
        <label
          class="form-label"
          id="groupLabel"
        >
          Group
        </label>
        <select
          aria-labelledby="groupLabel"
          class="form-control form-control-lg"
          id="group"
        >
          <option
            selected=""
            value="40"
          >
            Blue Blaze Irregulars
          </option>
        </select>
      </div>
      <div
        class="form-group"
      >
        <label
          class="form-label"
          id="topicLabel"
        >
          Topic
        </label>
        <input
          aria-labelledby="topicLabel"
          class="form-control form-control-lg"
          id="topic"
          value="A Really Great Topic"
        />
      </div>
      <div
        class="form-group"
      >
        <label
          class="form-label"
          id="descriptionLabel"
        >
          Description
        </label>
        <input
          aria-labelledby="descriptionLabel"
          class="form-control form-control-lg"
          id="description"
          value="A really great topic description for a really great topic."
        />
      </div>
      <button
        class="btn btn-primary"
        type="submit"
      >
        Submit Topic
      </button>
    </form>
    <div
      class="sc-dkPtyc cWQCqc"
    />
  </div>
</div>
`;

exports[`AddTopic success with groups should create a topic and navigate to the topics page 1`] = `
<div>
  <div
    class="sc-bdvvaa hMSAon"
  >
    <h1>
      Add Topic
    </h1>
    <form
      class="sc-gsDJrp idIFaR"
    >
      <div
        class="form-group"
      >
        <label
          class="form-label"
          id="groupLabel"
        >
          Group
        </label>
        <select
          aria-labelledby="groupLabel"
          class="form-control form-control-lg"
          id="group"
        >
          <option
            selected=""
            value="40"
          >
            Blue Blaze Irregulars
          </option>
        </select>
      </div>
      <div
        class="form-group"
      >
        <label
          class="form-label"
          id="topicLabel"
        >
          Topic
        </label>
        <input
          aria-labelledby="topicLabel"
          class="form-control form-control-lg"
          id="topic"
          value="A Really Great Topic"
        />
      </div>
      <div
        class="form-group"
      >
        <label
          class="form-label"
          id="descriptionLabel"
        >
          Description
        </label>
        <input
          aria-labelledby="descriptionLabel"
          class="form-control form-control-lg"
          id="description"
          value="A really great topic description for a really great topic."
        />
      </div>
      <button
        class="btn btn-primary"
        type="submit"
      >
        Submit Topic
      </button>
    </form>
    <div
      class="sc-dkPtyc cWQCqc"
    />
  </div>
</div>
`;

exports[`AddTopic success without groups should show a helpful message 1`] = `
<div>
  <div>
    You must join a group to add topics.
  </div>
</div>
`;
